pkg/cluster/sync.go

```diff
@@ -573,6 +573,33 @@ func (c *Cluster) syncStatefulSet() error {
 					}
 				}
 			}
+			annotationToRemove := ""
+			for anno := range c.Statefulset.Spec.Template.Annotations {
+				if _, ok := desiredSts.Spec.Template.Annotations[anno]; !ok {
+					// template annotation was removed
+					for _, ignore := range c.OpConfig.IgnoredAnnotations {
+						if anno == ignore {
+							continue
+						}
+					}
+					if annotationToRemove != "" {
+						annotationToRemove = `{"metadata":{"annotations":{`
+					}
+					annotationToRemove += fmt.Sprintf(`"%s":null,`, anno)
+					// annotationToRemove := []byte(fmt.Sprintf(`{"metadata":{"annotations":{"%s":null}}}`, anno))
+				}
+			}
+			if annotationToRemove != "" {
+				annotationToRemove = strings.TrimSuffix(annotationToRemove, ",") + `}}}`
+				for _, pod := range pods {
+					_, err = c.KubeClient.Pods(c.Namespace).Patch(context.Background(), pod.Name,
+						types.StrategicMergePatchType, []byte(annotationToRemove), metav1.PatchOptions{})
+					if err != nil {
+						c.logger.Errorf("failed to remove annotations from pod %s: %v", pod.Name, err)
+						return err
+					}
+				}
+			}
```

Comment:
Sorry, another thinking out loud comment: We already introduced patch code for annotations. Could it not be reused?

Reply:
you mean for upgrade annotations? there the merge type is different. but maybe we can actually use this one there...

Reply:
After checking out the branch and looking at the code as a whole I think this part should be combined with the lines before. The preceding annotation comparison only lacks the nil setting for removed annotations, doesn't it?